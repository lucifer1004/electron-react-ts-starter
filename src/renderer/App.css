h1 {
  text-align: center;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 3rem;
}
