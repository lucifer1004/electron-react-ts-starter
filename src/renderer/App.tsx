import React, {useState} from 'react'
import './App.css'
const App = () => {
  const [counter, setCounter] = useState(0)

  return (
    <>
      <h1>Hello TypeRelectron</h1>
      <p>{`You have clicked the button ${counter} time${
        counter === 1 ? '' : 's'
      }`}</p>
      <button onClick={() => setCounter(counter => counter + 1)}>Click</button>
    </>
  )
}

App.displayName = 'App'
export default App
