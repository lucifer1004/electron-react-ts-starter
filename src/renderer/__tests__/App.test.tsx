import React from 'react'
import {render, fireEvent} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import App from '../App'

describe('Test if', () => {
  it('App renders', () => {
    const {asFragment} = render(<App />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('Button works', () => {
    const {getByText, asFragment} = render(<App />)
    const p = getByText(/You have/)

    expect(p).toHaveTextContent('You have clicked the button 0 times')

    fireEvent.click(getByText('Click'))
    expect(p).toHaveTextContent('You have clicked the button 1 time')
    expect(asFragment()).toMatchSnapshot()

    fireEvent.click(getByText('Click'))
    expect(p).toHaveTextContent('You have clicked the button 2 times')
    expect(asFragment()).toMatchSnapshot()
  })
})
