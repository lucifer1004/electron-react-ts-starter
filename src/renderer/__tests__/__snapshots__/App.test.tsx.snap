// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if App renders 1`] = `
<DocumentFragment>
  <h1>
    Hello TypeRelectron
  </h1>
  <p>
    You have clicked the button 0 times
  </p>
  <button>
    Click
  </button>
</DocumentFragment>
`;

exports[`Test if Button works 1`] = `
<DocumentFragment>
  <h1>
    Hello TypeRelectron
  </h1>
  <p>
    You have clicked the button 1 time
  </p>
  <button>
    Click
  </button>
</DocumentFragment>
`;

exports[`Test if Button works 2`] = `
<DocumentFragment>
  <h1>
    Hello TypeRelectron
  </h1>
  <p>
    You have clicked the button 2 times
  </p>
  <button>
    Click
  </button>
</DocumentFragment>
`;
